packages/datadog-instrumentations/src/cassandra-driver.js

```diff
@@ -0,0 +1,207 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+addHook({ name: 'cassandra-driver', versions: ['>=3.0.0'] }, cassandra => {
+  const startCh = channel('apm:cassandra:batch:start')
+  const asyncEndCh = channel('apm:cassandra:batch:async-end')
+  const endCh = channel('apm:cassandra:batch:end')
+  const errorCh = channel('apm:cassandra:batch:error')
+
+  shimmer.wrap(cassandra.Client.prototype, 'batch', batch => function (queries, options, callback) {
+    const asyncResource = new AsyncResource('bound-anonymous-fn')
+    const query = combine(queries)
+    startCh.publish({ client: this, query })
```

Comment:
There should always be a check for `hasSubscribers` when a new object is created.

Reply:
Also, what is the needed information from `client`? Generally speaking, it makes supporting new versions easier to just grab what we need and not pass complex objects for which the interface might change.